api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	// If rules contains at least one Allow rule and none of them
```

Comment:
After chatting with @ZackButcher today, here are some suggestions to simplify this API
1. Default Deny -  if no rules are defined or no rules match, then `Deny`
```
authorization: {}
```

is same as 

```
authorization:
  rules:
  - action: Deny
```

is same as client IP being `3.3.3.3` and not matching any rules
```
authorization:
  rules:
  - clientSelectors:
       clientCIDRs:
        - 1.1.1.1/32  
     action: Deny
  - clientSelectors:
       clientCIDRs:
        - 2.2.2.2/32  
     action: Allow     
```

2. If Allow and Deny rules match, `Deny` takes precedence

Reply:
However 2. breaks this case - I'd like to deny the entire subnet `1.1.1.0/24` except `1.1.1.1`

Reply:
to solve above case, we use the `matcher_list` field https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/rbac/v3/rbac.proto#extensions-filters-http-rbac-v3-rbac and perform first match wins
cc @ZackButcher

Reply:
if we do first match, then for the below config
```
authorization:
  rules:
    - clientSelectors:
       methods:
        - POST  
     action: Deny
  - clientSelectors:
       clientCIDRs:
        - 1.1.1.1/32  
     action: Allow
  - clientSelectors:
       clientCIDRs:
        - 1.1.1.0/24  
     action: Deny
```

This would allow us to
* reject all `POST` requests
* allow client with IP `1.1.1.1` for other requests
* deny others

Reply:
in this case last deny is not needed. If the client ip does not match 1.1.1.1 its denied already before that

Reply:
agree, we are essentially using list order as an inherently priority order for rules